### PR TITLE
docs: On-demand compute docs

### PR DESCRIPTION
**Pull request by johnjcsmith**

### CODE REVIEW

File: cli/src/commands/deploy.ts [hunk 1]
```diff
@@ -14,5 +14,7 @@ export const Deploy: CommandModule = {
       .command(DeployList)
       .command(DeployInfo)
       .command(DeployLogs),
-  handler: DeployCreate.handler,
```
Comment (johnjcsmith) [inline cli/src/commands/deploy.ts:17]: Having `differential deploy` default to create could be unsafe (as there are sub-commands on `deploy` i.e `list`, `info`.

File: docs/deployments/on-demand-compute.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+On-demand compute is Differential's managed compute offering that runs your Differential services on-demand.
+
+Since the control-plane has the knowledge of how many function calls are being made and how frequently, it can use this information to spin up the service on-demand and scale it down when it's not being used.
+
+Because a single service session can execute multiple function calls, you don't have to worry about the cold start time of the function. The service will be kept warm for a certain period of time after the last function call.
+
+## Try the beta
+
+On-demand compute is currently in private beta. To gain early access, please sign up for the waitlist [here](https://forms.fillout.com/t/9M1VhL8Wxyus).
```
Comment (johnjcsmith) [inline docs/deployments/on-demand-compute.md:13]: This is the "cloud" wait list, not sure if we wan't a separate one or just ask people to get in touch by email?

File: docs/deployments/on-demand-compute.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
+# On-demand Compute (BETA)
+
+> On-demand compute is currently in private beta. To gain early access, please sign up for the waitlist [here](https://forms.fillout.com/t/9M1VhL8Wxyus).
```
Comment (johnjcsmith) [inline docs/deployments/on-demand-compute.md:3]: This is the "cloud" wait list, not sure if we wan't a separate one or just ask people to get in touch by email?

Comment (nadeesha) [reply docs/deployments/on-demand-compute.md:3]: Yeah, let's just do helo@differential.dev?

**Pull request merged**